### PR TITLE
UI: refactor light sensor

### PR DESCRIPTION
**Pull request by adeebshihadeh**
In preparation of #2147.

### CODE REVIEW

**Comment by deanlee** [comment]
I think it might not be as good as tthread.light sensor event is generated only if the value has changed,the performance of the threading will be much better than sensorEvents

**Comment by adeebshihadeh** [comment]
We're planning on implementing double tap to wake with the accelerometer soon, so this should be the same since the accelerometer updates faster than UI.

File: selfdrive/ui/android/ui.cc [hunk 1]
```diff
@@ -93,24 +41,35 @@ static void enable_event_processing(bool yes) {
...
+  static int display_mode = HWC_POWER_MODE_OFF;
+  static int display_timeout = 0;
+
+  // determine desired state
+  int desired_mode = display_mode;
+  if (user_input || s->ignition || s->started) {
+    desired_mode = HWC_POWER_MODE_NORMAL;
+    display_timeout = 30*UI_FREQ;
+  } else {
+    display_timeout = std::min(display_timeout-1, 0);
```
Comment (deanlee) [inline selfdrive/ui/android/ui.cc:56]: should be std::max

**Comment by pd0wm** [comment]
Branch became impossible to rebase. Squashed all changes into single commit.

**Pull request merged**